### PR TITLE
Add global-phase support and data plane token support

### PR DESCRIPTION
**Pull request by jobergum**
- Add global-phase support
- Add option for using token based authentication for data plane
- Simplify handling of session parameters

What is left:
- Integration tests need to be updated to test end to end.
- Add ability to read multiple endpoints for vespa cloud (mtls + token)

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by kkraune — DISMISSED** [review]
💯

File: vespa/application.py [hunk 1]
```diff
@@ -105,18 +106,20 @@ def __init__(
...
+        vespa_cloud_secret_token: Optional[str] = None,
         output_file: IO = sys.stdout,
         application_package: Optional[ApplicationPackage] = None,
     ) -> None:
         """
         Establish a connection with an existing Vespa application.
 
-        :param url: Vespa instance URL.
-        :param port: Vespa instance port.
+        :param url: Vespa data plane endpoint.
```
Comment (lesters) [inline vespa/application.py:116]: `data plane` is a bit confusing for users that want to connect to Vespa instances not on our cloud, for instance a locally running Docker instance?

Comment (lesters) [reply vespa/application.py:116]: This is just nit picking, but perhaps we should leave it as `endpoint` and `port` - and mention data planes etc in the param description below?

**Review by lesters — DISMISSED** [review]
Great!

File: vespa/application.py (hunk 1 above)
Comment (jobergum) [reply vespa/application.py:116]: Agree

Comment (jobergum) [reply vespa/application.py]: Fixed with https://github.com/vespa-engine/pyvespa/pull/573/commits/02ccd75b18f68950496e32ce897daf6b26e55854

**Review by kkraune — APPROVED** [review]
Awesome! 💯 x 💯

**Pull request merged**